app/services/password_resetter.rb

```diff
@@ -0,0 +1,11 @@
+class PasswordResetter
+
+  def initialize(user:)
+    @user = user
+  end
+
+  def generate_token
+    @user.update_attributes(password_token: SecureRandom.hex(8), password_token_generated_at: Time.now)
```

Comment:
8 might be little short. Sweet spot is around 12-14 - it's not visually bloated and it's really hard to forge.